### PR TITLE
Closes #227

### LINKED ISSUE #227: Rules NextBestNCRM` class cleanup
Please edit appropriately below if needed.

To do:

- [ ] Revise documentation
  - [ ] RStudio outline section dividers (use `# function or class name ----`)
  - [ ] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [ ] Replace `##'` by `#'`
  - [ ] Add to `pkgdown.yml`
  - [ ] Remove file history and the timestamp in comments on top of file
- [ ] Add unit tests
  - [ ] For classes:
    - [ ] Test default constructor
    - [ ] For reference classes: test methods
  - [ ] For functions: as usual
- [ ] Refactor as needed
  - [ ] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [ ] Keep externally used names so that we don't break downstream code
  - [ ] Remove `keywords` tags
  - [ ] For functions or methods:
    - [ ] Add assertions for arguments using `checkmate` functions
    - [ ] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [ ] Consider renaming internal variables etc. if needed
    - [ ] Avoid explicit use of `return`
  - [ ] For classes:
    - [ ] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [ ] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [ ] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #227

### CODE REVIEW

File: R/Rules-class.R [hunk 1]
```diff
@@ -78,74 +74,75 @@ NextBestMTD <- function(target, derive) {
...
 
-                     o$result()
-                 })
-validObject(.NextBestNCRM())
+#' `NextBestNCRM`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`NextBestNCRM`] is the class for next best dose that finds the next dose in
+#' target interval.
```
Comment (danielinteractive) [inline R/Rules-class.R:86]:
```suggestion
#' [`NextBestNCRM`] is the class for next best dose that finds the next dose with high posterior probability to be in the
#' target toxicity interval.
```

File: tests/testthat/test-Rules-class.R [hunk 2]
```diff
@@ -17,28 +17,22 @@ test_that("NextBestMTD object can be created with user constructor", {
   expect_valid(result, "NextBestMTD")
 })
 
-test_that("NextBestMTD throws the error for non valid target", {
```
Comment (danielinteractive) [inline tests/testthat/test-Rules-class.R:20]: what about those NextBestMTD tests? why did you remove those?

File: tests/testthat/test-Rules-validity.R [hunk 3]
```diff
@@ -29,6 +29,46 @@ test_that("v_next_best_mtd returns message for non-valid derive", {
...
+  # Changing `overdose` so that it is not an interval.
+  object@overdose <- 1.2
+  expect_equal(v_next_best_ncrm(object), "overdose has to be a probability range")
+
+  # Changing `overdose` so that the one bound is not a valid probability.
+  object@overdose <- c(0.4, 2)
+  expect_equal(v_next_best_ncrm(object), "overdose has to be a probability range")
+})
+
+test_that("h_next_best_ncrm returns message for non-valid max_overdose_prob", {
```
Comment (danielinteractive) [inline tests/testthat/test-Rules-validity.R:61]:
```suggestion
test_that("v_next_best_ncrm returns message for non-valid max_overdose_prob", {
```

**Review by danielinteractive — COMMENTED** [review]
thx Wojciech, please see inside, and also add the test(s) for the `nextBest()` method for this class in `test-Rules-methods.R`, thanks!

File: tests/testthat/test-Rules-class.R (hunk 2 above)
Comment (wwojciech) [reply tests/testthat/test-Rules-class.R:20]: Yes, I removed intentionally, as I realised these were a duplicated unit tests for the functionality that is provided by class validation function, and for this we already have unit tests.

**Review by danielinteractive — COMMENTED** [review]
Did you already add the test(s) for the nextBest() method?
